content/guides/getting-started/writing-your-first-component-test.md

```diff
@@ -2,4 +2,940 @@
+can write our first component test.
+
+Replace the contents of your spec file with this, and then save it:
+
+<code-group>
+<code-block label="React" active>
+
+```js
+import { mount } from '@cypress/react'
+import { LoginForm } from './LoginForm'
+
+it('should mount the component', () => {
+  mount(<LoginForm />)
+})
+```
+
+</code-block>
+<code-block label="Vue">
+
+```js
+import { mount } from '@cypress/vue'
+import { LoginForm } from './LoginForm'
+
+it('should mount the component', () => {
+  mount(<LoginForm />)
+})
+```
+
+</code-block>
+</code-group>
+
+Just like in the previous section, we should see one passing test. However, this
+time, because we're mounting a component, we should also see the component
+rendering in the Cypress app.
+
+(SCREENSHOT OF CYPRESS APP WITH MOUNTED COMPONENT)
+
+(link to mount docs ?)
+
+(mention getting components to work / render correctly ?)
+
+### React, Vue and JSX
+
+(explain how JS and JSX are supported in Cypress component tests, and how Vue
+template syntax is not supported, but how the mount command will work with both)
+
+## Testing the DOM
+
+Mounting a component in a test can be useful as a baseline assertion that a
+component doesn't error when mounted. However, we'll usually want to assert more
+specific things about the component.
+
+### Writing smart tests
+
+How can we assert that the password field has a `type` attribute of `password`,
+so we can be sure that the password is concealed as the user enters it?
+
+The [`cy.get()`](/api/commands/get) command both gets a DOM element from the
+rendered component, and implicitly asserts that it was found. So, we could do
+this to assert that there is an `input` element with a `type` attribute of
+`password`:
+
+```js
+it('should have password input', () => {
+  mount(<LoginForm />)
+  cy.get('input[type="password"]')
+})
+```
+
+However, this only asserts that there is an `input` element with a `type`
+attribute of `password` somewhere in the rendered component, which would give us
+false confidence that our component was working as-intended in the case where we
+accidentally swapped the username and password fields.
+
+Instead of trying to get the password field in a programmer-centric way, let's
+get the password field in a user-centric way. Instead of getting the `input`
+element by its attributes, let's get the `label` element by its text using
+[`cy.contains()`](/api/commands/contains) and then use
+[`cy.find()`](/api/commands/find) to find the descendant `input` element
+underneath it, which we will then assert has a `type` attribute of `password`
+using [`.should()`](/api/commands/should).
+
+After your last test, add this test and save the spec file:
+
+```js
+it('should have password input of type password', () => {
+  mount(<LoginForm />)
+  cy.contains('Password').find('input').should('have.attr', 'type', 'password')
+})
+```
+
+There are a few things about the Cypress UI to note:
```

Comment:
Curious what @amirrustam thinks about these "Cypress UI" notes. I'm afraid if I don't add them in, they'll be missed. But I can remove them if they're really superfluous.

Reply:
Generally, verbose information like this wouldn't be necessary, but this is a very introductory page so it is helpful considering the introductory audience. One point of improvement is to keep these as informational asides or helpful tips in a blue alert box, so the user sees that it's just a bit of side info.

Reply:
@cowboy also in this scenario when you say "Cypress UI" you are referring to the Cypress Command Log, so let's call it Command Log instead of UI, and we can linkify to https://deploy-preview-4201--cypress-docs.netlify.app/guides/core-concepts/test-runner#Command-Log